graphql-ballerina/tests/09_resources_returning_arrays.bal

```diff
@@ -222,3 +232,42 @@ isolated function testOptionalArrayInvalidQuery() returns error? {
     };
     assertJsonValuesWithOrder(result, expectedPayload);
 }
+
+@test:Config {
+    groups: ["array", "service", "unit"]
+}
+isolated function testResourceReturningServiceArrayObjectWithQueryReturnsErrors() returns error? {
+    string graphqlUrl = "http://localhost:9100/graphql";
+    string document = string `{ allVehicles { name registeredYear } }`;
+    json result = check getJsonPayloadFromService(graphqlUrl, document);
+    json expectedPayload = {
+        data: {
+            allVehicles: [
+                {
+                    name: "Benz",
+                    registeredYear: "2005"
+                },
+                {
+                    name: "BMW",
+                    registeredYear: "2010"
+                },
+                {
+                    name: "Ford"
+                }
+            ]
+        },
+        errors: [
+            {
+                message: "Registered Year is Not Found",
+                locations: [
+                    {
+                        line: 1,
+                        column: 22
+                    }
+                ],
+                path: ["allVehicles", 2, "registeredYear"]
+            }
+        ]
+    };
+    test:assertEquals(result, expectedPayload);
```

Comment:
Let's use the new utility function (`assertJsonValuesWithOrder`) to assert JSON payloads. This is because simple `assertEquals` does not consider the order of the JSON fields, but we have to maintain the order.